Tribler/Core/Libtorrent/LibtorrentMgr.py

```diff
@@ -633,17 +631,9 @@ def start_download(self, torrentfilename=None, infohash=None, tdef=None, dconfig
         # the priority of the parameters is: (1) tdef, (2) infohash, (3) torrent_file.
         # so if we have tdef, infohash and torrent_file will be ignored, and so on.
         if tdef is None:
-            if infohash is not None:
```

Comment:
This seems like it should stay.

Reply:
No, it can be safely removed. `get_collected_torrent` is long dead. The code block under this condition is basically never executed - and that's why we get no errors from the thing being dead.

Reply:
Agreed. Seems fine.

Reply:
Also, we should get rid of these nested `is None` or `is not None` combinations. There are methods that receive like four or five parameters and depending on whether they are set or not, the executed logic is completely different.